CurrencyConverterControl/Data/CurrencyDataProvider.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using CurrencyConverterControl.Model;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+
+namespace CurrencyConverterControl.Data
+{
+    internal class CurrencyDataProvider : ICurrencyDataProvider
+    {
+        string baseURl = "http://api.currencylayer.com/";
+        string access_key = "63da635cc4fff2e738aeb69cd8de0e91";
+        Dictionary<string, double> exchangeRates = new Dictionary<string, double>();
+
+        /// <summary>
+        /// This method is used to perfom the currency convertion
+        /// </summary>
+        /// <param name="currencyfrom">This parameter gets the value of the source currency</param>
+        /// <param name="currencyto">This parameter gets the value of the destination currency</param>
+        /// <param name="amount">This parameter gets the value which needs to be converted</param>
+        /// <returns></returns>
+        /// <exception cref="HttpRequestException"></exception>
+        public double Convert(string currencyfrom, string currencyto, double amount)
+        {
+            try
+            {
+                if (currencyfrom == currencyto) { return amount; }
+                using (HttpClient httpClient = new HttpClient())
```

Comment:
FYI, this is not a error, but it is better to use a static client or HttpClient factory. 
Get acquainted this guidlines: 
https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines

Reply:
@techNIKtos i researched a little bit on the IHttpClientFactory, which is in fact as you say it is better than HTTPclient, when l have to manage lifecycles or when l have multiple clients to manage. In this conditions l only have to make a simple call to the end point. 
Moreover based on what l read it needs to be injected into the startup.cs of a ASP.NET project, on which the project that this class is implemented is a .Net Custom Library, so l think l can not implement it. 
Please do let me know if my logic is wrong.